### PR TITLE
Remove redundant data in MountedContainer

### PR DESCRIPTION
**Pull request by alfonsoros88**
While working on the youki feature, noticed that `MountedContainer` had
some redundant information. This patch is a refactoring that tries to
remove the unnecessary `manifest` and `container` fields out of `MountedContainer`.

### CODE REVIEW

**Comment by flxo** [comment]
@alfonsoros88 I rebased because I merged #424

File: northstar/src/runtime/process/fs.rs [hunk 1]
```diff
@@ -92,35 +93,34 @@ impl Mount {
...
 
     for (target, mount) in manifest_mounts {
         match &mount {
             manifest::Mount::Bind(manifest::Bind { host, options }) => {
-                mounts.extend(bind(&root, target, host, options));
+                mounts.extend(bind(root, target, host, options));
             }
             manifest::Mount::Persist => {
-                mounts.push(persist(&root, target, config, container).await?);
+                let source = config.data_dir.join(manifest.name.to_string());
```
Comment (flxo) [inline northstar/src/runtime/process/fs.rs:112]: We shouldn't forget to document that the version is intentionally not part of the path here. This allows upgrades with persistent data migration.

File: northstar/src/runtime/process/fs.rs [hunk 2]
```diff
@@ -188,47 +188,49 @@ fn bind(root: &Path, target: &Path, host: &Path, options: &MountOptions) -> Vec<
 
 async fn persist(
     root: &Path,
+    source: &Path,
     target: &Path,
-    config: &Config,
-    container: &MountedContainer,
+    uid: u16,
+    gid: u16,
```
Comment (flxo) [inline northstar/src/runtime/process/fs.rs:194]: Like that!

File: northstar/src/runtime/state.rs [hunk 3]
```diff
@@ -841,4 +850,12 @@ impl<'a> State<'a> {
                 .expect("Internal channel error on main");
         }
     }
+
+    /// Returns an iterator over the containers currently running
+    fn running_containers(&self) -> impl Iterator<Item = &Container> {
+        self.containers
+            .iter()
+            .filter(|(_, state)| state.process.is_some())
```
Comment (flxo) [inline northstar/src/runtime/state.rs:857]: nit: There's a handy [filter_map](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map) and I think this should work:
```rust
.filter_map(|(container, state)| state.proces.map(|_| container))
```

**Review by flxo — APPROVED** [review]
LGTM!

**Comment by alfonsoros88** [comment]
> @alfonsoros88 I rebased because I merged #424

whoops... didn't see this on time :see_no_evil:

**Comment by flxo** [comment]
> > @alfonsoros88 I rebased because I merged #424
> 
> whoops... didn't see this on time 🙈

No worries - feel free to discard it...:-)

File: northstar/src/runtime/state.rs (hunk 3 above)
Comment (alfonsoros88) [reply northstar/src/runtime/state.rs]: I wasn't completely happy using either `state.proces.map(|_| container)` or ``state.proces.and(Some(container))`, it reads kind of weird. Fortunately `rust-analyzer` saved the day :tada: and suggested `state.process.is_some().then(|| container)` that is clearer and nicer to read. Didn't know those `.then` and `.then_some` methods on `bool` :)

File: northstar/src/runtime/process/fs.rs (hunk 1 above)
Comment (alfonsoros88) [reply northstar/src/runtime/process/fs.rs:114]: Not sure I understand fully this but added your comment in a note above this line :blush:

**Pull request merged**